specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json

```diff
@@ -1013,9 +1013,472 @@
+        "description": "Verifies the certificate's private key posession by providing the leaf cert issued by the verifying pre uploaded certificate.",
+        "operationId": "Certificates_Verify",
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The name of the resource group that contains the IoT hub.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "description": "The name of the IoT hub.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "certificateName",
+            "in":"path",
+            "description":"The name of the certificate",
+            "required":true,
+            "type":"string"
+          },
+          {
+            "name": "certificateVerificationBody",
+            "in":"body",
+            "description":"The name of the certificate",
+            "required":true,
+            "schema": {
+              "$ref": "#/definitions/CertificateVerificationDescription"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The body contains the certificate.",
+            "schema": {
+              "$ref": "#/definitions/CertificateDescription"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
     }
   },
   "definitions": {
+	"CertificateVerificationDescription":{
+      "description": "The JSON-serialized leaf certificate",
+      "type": "object",
+      "properties": {
+        "certificate": {
+          "description": "base-64 representation of X509 certificate .cer file or just .pem file content.",
+          "type": "string"
+        }
+      }
+    },
+    "CertificateBodyDescription":{
+      "description": "The JSON-serialized X509 Certificate.",
+      "type": "object",
+      "properties": {
+        "certificate": {
+          "description": "base-64 representation of the X509 leaf certificate .cer file or just .pem file content.",
+          "type": "string"
+        }
+      }
+    },
+    "CertificateListDescription": {
+      "description": "The JSON-serialized array of Certificate objects.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The array of Certificate objects.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CertificateDescription"
+          }
+        }
+      }
+    },
+    "CertificateDescription": {
```

Comment:
Why there is no `id` field for this model `CertificateDescription`? It looks like it can have id with standard ARM format `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/certificates/{certificateName}`


[1]. With id, consumer should be able to extract resource group name, parent iot hub name. This can be further used to perform update or delete operation on a certificate instance, because those actions required these fields.